include/targets/hip/load_store.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <register_traits.h>
+
+namespace quda
+{
+
+  // pre-declaration of vector_load that we wish to specialize
+  template <bool> struct vector_load_impl;
+
+  // CUDA specializations of the vector_load
+  template <> struct vector_load_impl<true> {
+    template <typename T> __device__ inline void operator()(T &value, const void *ptr, int idx)
+    {
+      value = reinterpret_cast<const T *>(ptr)[idx];
+    }
+
+    __device__ inline void operator()(short8 &value, const void *ptr, int idx)
+    {
+      float4 tmp;
+      operator()(tmp, ptr, idx);
+      memcpy(&value, &tmp, sizeof(float4));
+    }
+
+    __device__ inline void operator()(char8 &value, const void *ptr, int idx)
+    {
+      float2 tmp;
+      operator()(tmp, ptr, idx);
+      memcpy(&value, &tmp, sizeof(float2));
+    }
+  };
+
+  // pre-declaration of vector_store that we wish to specialize
+  template <bool> struct vector_store_impl;
+
+  // CUDA specializations of the vector_store using inline ptx
+  template <> struct vector_store_impl<true> {
+    template <typename T> __device__ inline void operator()(void *ptr, int idx, const T &value)
+    {
+      reinterpret_cast<T *>(ptr)[idx] = value;
+    }
+
+#if 0
```

Comment:
Why not just delete these lines rather than hide them behind `#if 0`?

Reply:
I think I was keeping these as a placeholder in case I ever found out if hipcc implements them or not. I can remove them.

Reply:
These are calls to inline ptx functions that I wrote, so I think it unlikely that hipcc support them 😄 .  Not really needed for CUDA either TBH.

Reply:
ifdef 0-d cpde removed in 4a79b7